src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -2640,21 +2640,27 @@ public void ReceiverParameter_Out_01()
         var src = """
 public static class Extensions
 {
-    extension(out int i) { }
+    extension(out int i)
+    {
+        void M2() { }
+    }
     static void M(this out int i) { }
 }
 """;
         var comp = CreateCompilation(src);
         comp.VerifyEmitDiagnostics(
             // (3,15): error CS8328:  The parameter modifier 'out' cannot be used with 'extension'
-            //     extension(out int i) { }
+            //     extension(out int i)
             Diagnostic(ErrorCode.ERR_BadParameterModifiers, "out").WithArguments("out", "extension").WithLocation(3, 15),
-            // (4,17): error CS0177: The out parameter 'i' must be assigned to before control leaves the current method
+            // (5,14): error CS0177: The out parameter 'i' must be assigned to before control leaves the current method
```

Comment:
>// (5,14): error CS0177: The out parameter 'i' must be assigned to before control leaves the current metho [](http://example.com/codeflow?start=12&length=106)

Is this error going to be reported for a static method?

Reply:
It should not. We only enter/leave the extension parameter for instance methods. I'll add a test